tests/cli/main.cpp

```diff
@@ -197,252 +197,218 @@ class client_connection
-      // save the private key for this new account in the wallet file
-      BOOST_CHECK(con.wallet_api_ptr->import_key("jmjatlanta", bki.wif_priv_key));
-      con.wallet_api_ptr->save_wallet_file(con.wallet_filename);
-
-      // attempt to give jmjatlanta some bitsahres
-      BOOST_TEST_MESSAGE("Transferring bitshares from Nathan to jmjatlanta");
-      signed_transaction transfer_tx = con.wallet_api_ptr->transfer("nathan", "jmjatlanta", "10000", "1.3.0", "Here are some CORE token for your new account", true);
-
-      // get the details for init1
-      witness_object init1_obj = con.wallet_api_ptr->get_witness("init1");
-      int init1_start_votes = init1_obj.total_votes;
-      // Vote for a witness
-      signed_transaction vote_witness1_tx = con.wallet_api_ptr->vote_for_witness("jmjatlanta", "init1", true, true);
-
-      // generate a block to get things started
-      BOOST_CHECK(generate_block(app1));
-      // wait for a maintenance interval
-      BOOST_CHECK(generate_maintenance_block(app1));
-
-      // Verify that the vote is there
-      init1_obj = con.wallet_api_ptr->get_witness("init1");
-      witness_object init2_obj = con.wallet_api_ptr->get_witness("init2");
-      int init1_middle_votes = init1_obj.total_votes;
-      BOOST_CHECK(init1_middle_votes > init1_start_votes);
-
-      // Vote for a 2nd witness
-      int init2_start_votes = init2_obj.total_votes;
-      signed_transaction vote_witness2_tx = con.wallet_api_ptr->vote_for_witness("jmjatlanta", "init2", true, true);
-
-      // send another block to trigger maintenance interval
-      BOOST_CHECK(generate_maintenance_block(app1));
-
-      // Verify that both the first vote and the 2nd are there
-      init2_obj = con.wallet_api_ptr->get_witness("init2");
-      init1_obj = con.wallet_api_ptr->get_witness("init1");
-
-      int init2_middle_votes = init2_obj.total_votes;
-      BOOST_CHECK(init2_middle_votes > init2_start_votes);
-      int init1_last_votes = init1_obj.total_votes;
-      BOOST_CHECK(init1_last_votes > init1_start_votes);
-
-      // wait for everything to finish up
-      fc::usleep(fc::seconds(1));
-   } catch( fc::exception& e ) {
-      edump((e.to_detail_string()));
-      throw;
-   }
-   app1->shutdown();
+    try
+    {
+        BOOST_TEST_MESSAGE("Cli Vote Test for 2 Witnesses");
+
+        account_object nathan_acct_before_upgrade, nathan_acct_after_upgrade;
+        std::vector<signed_transaction> import_txs;
+        signed_transaction upgrade_tx, transfer_tx;
+
+        BOOST_TEST_MESSAGE("Importing nathan's balance");
+        import_txs = con->wallet_api_ptr->import_balance("nathan", nathan_keys, true);
+        nathan_acct_before_upgrade = con->wallet_api_ptr->get_account("nathan");
+
+        // upgrade nathan
+        BOOST_TEST_MESSAGE("Upgrading Nathan to LTM");
+        upgrade_tx = con->wallet_api_ptr->upgrade_account("nathan", true);
+        nathan_acct_after_upgrade = con->wallet_api_ptr->get_account("nathan");
```

Comment:
The upgrade of nathan account can be abstracted to a single test case, then be invoked by other test cases. See this for example: https://github.com/bitshares/bitshares-core/blob/610dbf519aa2a97523a6d753df93a00f368e4f8e/tests/tests/operation_tests.cpp#L983

Reply:
will do it soon

Reply:
INVOKE doesn't work well enough with test fixtures, also need to fight here with errors